src/potions.ts

```diff
@@ -348,6 +382,117 @@ function useAsValuable(potion: Potion, embezzlers: number, embezzlersOnly: boole
   return total;
 }
 
+const toothpickPotion = new Potion($item`papier-mâché toothpicks`, {
+  price: (historical: boolean) => {
+    return (
+      4 *
+      (npcPrice($item`soda water`) +
+        (historical && historicalAge($item`Rad Lib`) < 14
+          ? historicalPrice($item`Rad Lib`)
+          : mallPrice($item`Rad Lib`)))
+    );
+  },
+  acquire: (
+    qty: number,
+    item: Item,
+    maxPrice?: number | undefined,
+    throwOnFail?: boolean,
+    maxAggregateCost?: number | undefined,
+    tryRetrievingUntradeable?: boolean
+  ) => {
+    const globs = acquire(
+      4 * qty,
+      $item`Rad Lib`,
+      maxPrice ? maxPrice / 2 : mallPrice($item`Rad Lib`) * 2,
+      throwOnFail ?? false,
+      maxAggregateCost ?? 50000,
+      tryRetrievingUntradeable ?? false
+    );
+    Array(Math.floor(globs / 4))
+      .fill(0)
+      .every(() => create(item));
+    return itemAmount(item);
+  },
+});
+
+const rufusPotion = new Potion($item`closed-circuit pay phone`, {
+  providesDoubleDuration: false,
+  canDouble: false,
+  effect: $effect`Shadow Waters`,
+  duration: 30,
+  price: (historical: boolean) => {
+    const canAcquireItemQuest = rufusTarget() === null;
```

Comment:
bind the target to a variable; it'll make the nullish checks and type assertions much easier